newrelic/api/opentelemetry.py

```diff
@@ -208,7 +233,13 @@ def update_name(self, name):
             self.nr_trace.name = self._name
 
     def is_recording(self):
-        return self._sampled() and not (getattr(self.nr_trace, None), "end_time", None)
+        # TODO: Refine this logic further if needed.
+        if getattr(self.nr_trace, "end_time", None):
```

Comment:
```suggestion
        # If the trace has an end time set then it is done recording. Otherwise, if it does not have an end time set and the transaction's priority has not been set yet or it is set to something other than 0 then it is also still recording.
        if getattr(self.nr_trace, "end_time", None):
```